### PR TITLE
Bleck's refactor

### PR DESCRIPTION
**Pull request by OverlordBleck**
🗡

### CODE REVIEW

**Comment by OverlordBleck** [comment]
I don't know if i did this horribly wrong or not, please be gentle. ❤️

File: lua/autorun/server/sv_autoclean.lua [hunk 1]
```diff
@@ -1,36 +1,45 @@
...
 	
-	for k, v in pairs( ents.GetAll() ) do
-			if v:IsWeapon() and not (table.HasValue(player_weps,v) or (table.HasValue(turrets,v))) then
-				count = count + 1
-				v:Remove()
-			end
+	runCleanupCommandsOnPlayers()
+
+	for _, entity in pairs( ents.GetAll() ) do
+		if not IsValid( entity ) then continue end
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_autoclean.lua]: Why not 
```lua 
if not ( IsValid( entity ) and IsValid( entity.Owner ) ) then return end

local classIsBlacklisted = BlacklistedEntityTypes[entity:GetClass()]

if classIsBlacklisted or entity:isWeapon() then continue end

Comment (OverlordBleck) [reply lua/autorun/server/sv_autoclean.lua]: ENT.Owner only works for weapons IIRC

Comment (OverlordBleck) [reply lua/autorun/server/sv_autoclean.lua]: And the way you suggested just checks if it is a weapon then skips it, it wont remove grounded weapons.
ENT.Owner gets the current wielder of the weapon.

**Comment by OverlordBleck** [comment]
Oops... My preferences fucked up and got rid of spaces for tabs :(

File: lua/autorun/server/sv_autoclean.lua (hunk 1 above)
Comment (justinmuskopf) [reply lua/autorun/server/sv_autoclean.lua]: Okay, here's another stab at it, then.

Also, I remember something about lua/glua not having the increment operators `+=` or `++`.

Can you look into that?

```lua
if not IsValid( entity ) then continue end

local classIsBlacklisted = BlacklistedEntityTypes[entity:GetClass()]
local isOwnedWeapon = entity:IsWeapon() and IsValid( entity.Owner )

if isOwnedWeapon or not classIsBlacklisted then continue end

entity:Remove()

removedEntities = removedEntities + 1
```

Comment (OverlordBleck) [reply lua/autorun/server/sv_autoclean.lua]: I can do that for sure

File: lua/autorun/server/sv_autoclean.lua [hunk 2]
```diff
@@ -1,36 +1,57 @@
...
+
+        removedCount = removedCount + 1
+        entity:Remove()
+    end
+    
+    MsgAll("[CFC_Autoclean] Removed " .. tostring( removedCount ) .. " objects.")
 end
 
-timer.Create('cfcautoclean',GetConVar('cfc_autoclean'):GetInt(),0,cfcCleanServer)
+timer.Create("cfc_autoclean", GetConVar("cfc_autoclean"):GetInt(), 0, cfcCleanServer)
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_autoclean.lua]: Following convention:

```suggestion
timer.Create("cfc_AutoClean", GetConVar("cfc_autoclean"):GetInt(), 0, cfcCleanServer)
```

File: lua/autorun/server/sv_autoclean.lua [hunk 3]
```diff
@@ -1,36 +1,57 @@
...
+    ["gmod_turret"] = true
+}
+
+local CleanupCommands = {
+    ["r_cleardecals"] = true,
+    ["stopsound"] = true
+}
+
+local function notifyPlayers( count )
+    local Message = "[CFC_Autoclean] Removed " .. tostring(count) .. "objects."
```
Comment (brandonsturgeon) [inline lua/autorun/server/sv_autoclean.lua]: Can this var be lowercase instead

File: lua/autorun/server/sv_autoclean.lua [hunk 4]
```diff
@@ -1,36 +1,57 @@
-	CreateConVar('cfc_autoclean','500',{FCVAR_ARCHIVE, FCVAR_NOTIFY },'Autocleans the server based on seconds given')
-	
+
+local ConVarFlags = {FCVAR_ARCHIVE, FCVAR_NOTIFY}
+local DefaultCleanIntervalInSeconds = "500"
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_autoclean.lua]: DEFAULT_CLEAN_INTERVAL_IN_SECONDS

File: lua/autorun/server/sv_autoclean.lua [hunk 5]
```diff
@@ -1,36 +1,57 @@
...
+    ["gmod_wire_turret"] = true,
+    ["gmod_turret"] = true
+}
+
+local CleanupCommands = {
+    ["r_cleardecals"] = true,
+    ["stopsound"] = true
+}
+
+local function notifyPlayers( count )
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_autoclean.lua]: @brandonsturgeon are we wanting to chat print this to plys?

File: lua/autorun/server/sv_autoclean.lua [hunk 6]
```diff
@@ -1,36 +1,57 @@
...
+    local message = "[CFC_Autoclean] Removed " .. tostring(count) .. "objects."
+
+    print(Message) -- and server
+    for _, ply in pairs( player.GetHumans() ) do
+        ply:ChatPrint(Message)
```
Comment (justinmuskopf) [inline lua/autorun/server/sv_autoclean.lua]: @brandonsturgeon are we wanting to chat print this to plys?

Comment (brandonsturgeon) [reply lua/autorun/server/sv_autoclean.lua]: What are your thoughts?

Comment (OverlordBleck) [reply lua/autorun/server/sv_autoclean.lua]: I thought it was printed to the chat anyways...?

Comment (justinmuskopf) [reply lua/autorun/server/sv_autoclean.lua]: I don't see harm in it, personally.

**Pull request merged**